Assets/LeapMotion/Scripts/DetectionUtilities/ExtendedFingerDetector.cs

```diff
@@ -0,0 +1,110 @@
+﻿using UnityEngine;
+using System.Collections;
+using Leap;
+
+namespace Leap.Unity {
+
+  public class ExtendedFingerDetector : Detector {
+    [Tooltip("The interval in seconds at which to check this detector's conditions.")]
+    public float Period = .1f; //seconds
+
+    public IHandModel handModel = null;
+  
+    public PointingState Thumb = PointingState.Either;
+    public PointingState Index = PointingState.Either;
+    public PointingState Middle = PointingState.Either;
+    public PointingState Ring = PointingState.Either;
+    public PointingState Pinky = PointingState.Either;
+  
+    void Start () {
+      if(handModel == null){
+        handModel = gameObject.GetComponentInParent<IHandModel>();
+      }
+    }
+  
+    void OnEnable () {
+      StartCoroutine(extendedFingerWatcher());
+    }
+  
+    void OnDisable () {
+      StopCoroutine(extendedFingerWatcher());
+      Deactivate();
+    }
+  
+    IEnumerator extendedFingerWatcher() {
+      Hand hand;
+      while(true){
+        bool fingerState = false;
+        if(handModel != null && handModel.IsTracked){
+          hand = handModel.GetLeapHand();
+          if(hand != null){
+            fingerState = matchFingerState(hand.Fingers[0], 0)
+              && matchFingerState(hand.Fingers[1], 1)
+              && matchFingerState(hand.Fingers[2], 2)
+              && matchFingerState(hand.Fingers[3], 3)
+              && matchFingerState(hand.Fingers[4], 4);
+            if(handModel.IsTracked && fingerState){
+              Activate();
+            } else if(!handModel.IsTracked || !fingerState) {
+              Deactivate();
+            }
+          }
+        } else if(IsActive){
+          Deactivate();
+        }
+        yield return new WaitForSeconds(Period);
+      }
+    }
+  
+    private bool matchFingerState(Finger finger, int ordinal){
+      switch(ordinal){
```

Comment:
This could be simplified so that the switch only selects which PointingState to use, and then the conditional logic doesn't have to be repeated.